VirtoCommerce.CartModule.Data/Handlers/DeletePropertyCartChangedEventHandler.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.Domain.Cart.Events;
+using VirtoCommerce.Domain.Cart.Model;
+using VirtoCommerce.Domain.Common.Events;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.DynamicProperties;
+using VirtoCommerce.Platform.Core.Events;
+
+namespace VirtoCommerce.CartModule.Data.Handlers
+{
+    public class DeletePropertyCartChangedEventHandler : IEventHandler<CartChangedEvent>
+    {
+        private readonly IDynamicPropertyService _dynamicPropertyService;
+
+        public DeletePropertyCartChangedEventHandler(IDynamicPropertyService dynamicPropertyService)
+        {
+            _dynamicPropertyService = dynamicPropertyService;
+        }
+
+        public Task Handle(CartChangedEvent message)
+        {
+                foreach (var changedEntry in message.ChangedEntries)
+                {
+                   if(changedEntry.NewEntry.Items.Count() < changedEntry.OldEntry.Items.Count())
+                       TryDeleteCartProperty(changedEntry);
+                }
+            return Task.CompletedTask;
+        }
+
+        protected virtual void TryDeleteCartProperty(GenericChangedEntry<ShoppingCart> changedEntry)
+        {
+            var shoppingCart = changedEntry.OldEntry;
+
+            var removedLineItemProperties = new List<DynamicObjectProperty>();
+
+            var changedLineItems = changedEntry.NewEntry.Items.ToArray();
```

Comment:
Why didn't you use `CompareTo` extensions methods for cart line items changes detection?